extensions/amp-story-dev-tools/0.1/amp-story-dev-tools.js

```diff
@@ -32,21 +39,185 @@ const fontsToLoad = [
+            <path
+              d="M9.9165 4.08333L9.094 4.90583L10.599 6.41667H4.6665V7.58333H10.599L9.094 9.08833L9.9165 9.91667L12.8332 7L9.9165 4.08333ZM2.33317 2.91667H6.99984V1.75H2.33317C1.6915 1.75 1.1665 2.275 1.1665 2.91667V11.0833C1.1665 11.725 1.6915 12.25 2.33317 12.25H6.99984V11.0833H2.33317V2.91667Z"
+              fill="black"
+            />
+          </svg>
+        </div>
+      </div>
+      <div class="i-amphtml-story-dev-tools-tab">
+        <div class="lds-dual-ring"></div>
+      </div>
+    </div>
+  `;
+};
+
+/** @enum {string} */
+const DevToolsTab = {
+  PREVIEW: 'Preview',
+  PAGE_EXPERIENCE: 'Page Experience',
+  ERRORS: 'Errors',
+};
+
 export class AmpStoryDevTools extends AMP.BaseElement {
   /** @param {!AmpElement} element */
   constructor(element) {
     super(element);
+
+    const hashParams = parseQueryString(this.win.location.hash);
+
+    this.win.document.title = `Story Dev-Tools (${this.win.document.title})`;
+
+    // TODO: Remove support for url queryParam when widely available.
+    /** @private {string} */
+    this.storyUrl_ = hashParams['url'] || this.win.location.href.split('#')[0];
+
+    /** @private {!DevToolsTab} get URL param for tab (eg: #tab=page-experience) or default to PREVIEW*/
+    this.currentTab_ =
+      Object.values(DevToolsTab).find(
+        (e) => e.toLowerCase().replace(' ', '-') === hashParams['tab']
+      ) || DevToolsTab.PREVIEW;
+
+    /** @private {!Object} maps tabs to contents */
+    this.tabContents_ = {
+      [DevToolsTab.PREVIEW]: new AmpStoryDevToolsTab(
+        createTabContentsTemplate(this.element, DevToolsTab.PREVIEW),
+        this.win,
+        this,
+        this.storyUrl_
+      ),
+      [DevToolsTab.ERRORS]: new AmpStoryDevToolsTab(
+        createTabContentsTemplate(this.element, DevToolsTab.ERRORS),
+        this.win,
+        this,
+        this.storyUrl_
+      ),
+      [DevToolsTab.PAGE_EXPERIENCE]: new AmpStoryDevToolsTab(
+        createTabContentsTemplate(this.element, DevToolsTab.PAGE_EXPERIENCE),
+        this.win,
+        this,
+        this.storyUrl_
+      ),
+    };
   }
 
   /** @override */
   buildCallback() {
     this.loadFonts_();
 
-    // TODO(mszylkowski): Create actual layout.
-    const message = this.element.ownerDocument.createElement('div');
-    message.textContent = 'AMP Story Dev-tools was loaded, page WIP';
-    this.element.textContent = '';
-    this.element.appendChild(message);
+    this.buildLayout_();
+  }
+
+  /**
+   * @private
+   */
+  buildLayout_() {
+    const container = buildContainerTemplate(this.element);
+    this.element.appendChild(container);
+    this.element
+      .querySelector('.i-amphtml-story-dev-tools-close')
+      .addEventListener('click', () => {
+        this.win.location.href = this.storyUrl_;
+      });
+
+    // Create tabs on top
+    const tabsContainer = container.querySelector(
+      '.i-amphtml-story-dev-tools-tabs'
+    );
+    Object.values(DevToolsTab).forEach((e) => {
```

Comment:
nit: At first glance I thought `e` for `event`.
I wonder if `e` could be something else like `tab`.

Reply:
Sounds good, `tab` is currently the actual span used for the tab but sth like `tabTitle` is better